File: ios/Classes/session/SessionHolder.swift
import Adyen

class SessionHolder {
    var session: AdyenSession?
    var sessionDelegate: AdyenSessionDelegate?
    var adyenCheckout : AdyenCheckout?

    func setup(
        session: AdyenSession,
        sessionDelegate: AdyenSessionDelegate
    ) {
        self.session = session
        self.sessionDelegate = sessionDelegate
    }

    func reset() {
        session = nil
        sessionDelegate = nil
        adyenCheckout = nil
    }
}

File: android/src/main/kotlin/com/adyen/checkout/flutter/session/SessionHolder.kt
package com.adyen.checkout.flutter.session

import com.adyen.checkout.core.common.CheckoutContext
import org.json.JSONObject

class SessionHolder(
    var sessionSetupResponse: JSONObject = JSONObject(),
    var sessionCheckout: CheckoutContext.Sessions? = null
) {
    fun reset() {
        sessionSetupResponse = JSONObject()
    }
}
